### PR TITLE
James-3081 Make messages persistent in rabbitmq mailqueue

### PR DESCRIPTION
**Pull request by remk**
Regarding the persistence of messages in rabbitmq : To achieve this there are two steps :

the queue must be declared as durable (the declaration of the queue is persisted rabbitmq is restarted)
when sending a message, it must have a property setting its delivery_mode to persistent. (the messages with this property are persisted in case of a restart of rabbitmq)

At the time being the rabbitmq mailqueue in james :

the queue is declared durable => Good
the delivery_mode property is not set => Not good
the exchange is  durable =>  Good

What should be done is :

set the `delivery_mode` property to '2' ( see https://www.rabbitmq.com/releases/rabbitmq-java-client/v2.4.1/rabbitmq-java-client-javadoc-2.4.1/index.html?com/rabbitmq/client/MessageProperties.html )

### CODE REVIEW

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java [hunk 1]
```diff
@@ -250,6 +252,45 @@ void dequeueShouldWorkAfterNetworkOutage() throws Exception {
...
+            getMailQueue().enQueue(defaultMail()
+                .name(name3)
+                .build());
+
+            Thread.sleep(2000);
+            restartRabbitMQ();
+
+
+
+            List<MailQueue.MailQueueItem> items = dequeueFlux.take(3).collectList().block(Duration.ofSeconds(10));
```
Comment (chibenwa) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java]: Why so much line breaks?

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java [hunk 2]
```diff
@@ -250,6 +252,45 @@ void dequeueShouldWorkAfterNetworkOutage() throws Exception {
...
+
+            getMailQueue().enQueue(defaultMail()
+                .name(name2)
+                .build());
+
+            getMailQueue().enQueue(defaultMail()
+                .name(name3)
+                .build());
+
+            Thread.sleep(2000);
```
Comment (chibenwa) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java]: I'm ok waiting but can you add a comment explaining what we are waiting for?

Also, I've been seeing some rabbit restart without a "wait" before them. Maybe that's some potential bugs?

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java [hunk 3]
```diff
@@ -250,6 +252,45 @@ void dequeueShouldWorkAfterNetworkOutage() throws Exception {
...
+
+
+            List<MailQueue.MailQueueItem> items = dequeueFlux.take(3).collectList().block(Duration.ofSeconds(10));
+
+            assertThat(items)
+                .extracting(item -> item.getMail().getName())
+                .containsExactly(name1, name2, name3);
+        }
+
+        private void restartRabbitMQ() throws Exception {
```
Comment (chibenwa) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java]: Shouldn't this be in rabbitMQExtension ?

Comment (remk) [reply server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java]: Yes, that's what i am thinking now that i work on the event bus PR.
Which lead me to the issue that the `DockerRabbitMQ.restart` method used in the `RabbitMQEventBustTest` do nothing. So all the tests pass but in reality nothing work as expected ... 
I will create an issue for this

**Comment by Arsnael** [comment]
```
[8670da7be38cf7a362c4074c1b0a6b8cfdc14571] [ERROR] Errors: 
[8670da7be38cf7a362c4074c1b0a6b8cfdc14571] [ERROR]   MessageFullViewFactoryTest$WithProjectionInvolvedTest.fromMessageResultsShouldComputeWhenProjectionReturnEmpty:883 » NullPointer
```
second build I see that... I will check locally

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java (hunk 3 above)
Comment (remk) [reply server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java]: #3165

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java (hunk 2 above)
Comment (remk) [reply server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java]: I though that the enqueue was  an async process. But it is synchronous in fact. I removed the sleep.

**Comment by chibenwa** [comment]
Ok we merge this anyway.

Remove the label if you have the time to run load testing on this.

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**